packages/camera/example/lib/main.dart

```diff
@@ -24,149 +26,316 @@ IconData cameraLensIcon(CameraLensDirection direction) {
-      floatingActionButton: (controller == null)
-          ? null
-          : new FloatingActionButton(
-              child: const Icon(Icons.camera),
-              onPressed: controller.value.isStarted ? capture : null,
-            ),
-    );
   }
 
-  Widget imageWidget() {
-    return new Expanded(
-      child: new Align(
-        alignment: Alignment.centerRight,
-        child: new SizedBox(
-          child: new Image.file(new File(imagePath)),
-          width: 64.0,
-          height: 64.0,
-        ),
-      ),
-    );
+  void onTakePictureButtonPressed() {
+    takePicture().then((String filePath) {
+      if (mounted) {
+        setState(() {
+          imagePath = filePath;
+          videoController?.dispose();
+          videoController = null;
+        });
+        showInSnackBar('Picture saved to $filePath');
+      }
+    });
   }
 
-  Widget playPauseButton() {
-    return new FlatButton(
-      onPressed: () {
-        setState(
-          () {
-            if (controller.value.isStarted) {
-              controller.stop();
-            } else {
-              controller.start();
-            }
-          },
-        );
-      },
-      child:
-          new Icon(controller.value.isStarted ? Icons.pause : Icons.play_arrow),
-    );
+  void onVideoRecordButtonPressed() {
+    startVideoRecording().then((String filePath) {
+      if (mounted) setState(() {});
+      showInSnackBar('Saving video to $filePath');
+    });
   }
 
-  Future<Null> capture() async {
-    if (controller.value.isStarted) {
-      final Directory tempDir = await getTemporaryDirectory();
-      if (!mounted) {
-        return;
+  Future<void> onStopButtonPressed() async {
+    await stopVideoRecording();
+    if (!mounted) {
+      return null;
+    }
+    showInSnackBar('Video recorded to: $videoPath');
+  }
+
+  Future<String> startVideoRecording() async {
+    if (!controller.value.isInitialized) {
+      showInSnackBar('Error: select a camera first.');
+      return null;
+    }
+    final Directory extDir = await getApplicationDocumentsDirectory();
+    final String dirPath = '${extDir.path}/Movies/flutter_test';
+    await new Directory(dirPath).create(recursive: true);
+    final String filePath = '$dirPath/${timestamp()}.mp4';
+    try {
+      videoPath = filePath;
+      await controller.startVideoRecording(filePath);
+    } on CameraException catch (e) {
+      logError(e.code, e.description);
+    }
+    return filePath;
+  }
+
+  Future<void> stopVideoRecording() async {
+    if (controller.value.isRecordingVideo) {
+      try {
+        await controller.stopVideoRecording();
+      } on CameraException catch (e) {
+        logError(e.code, e.description);
       }
-      final String tempPath = tempDir.path;
-      final String path = '$tempPath/picture${pictureCount++}.jpg';
-      await controller.capture(path);
+      final VideoPlayerController vcontroller =
+          new VideoPlayerController.network('file://$videoPath');
```

Comment:
We just added a .file() constructor  to the VideoPlayerController - we should use that.

Reply:
Done